docs/user-guide/query/gaffer-syntax/faqs.md

```diff
@@ -0,0 +1,184 @@
+# FAQs
+
+These are some previously asked questions about querying in Gaffer.
+
+## Iterable results
+
+**If I do queries such as `GetElements` or `GetAdjacentIds` the response type is an `Iterable` - why?**
+
+To avoid loading all query results into memory, [Gaffer Stores](../../../administration-guide/gaffer-stores/store-guide.md)
+should return an [`Iterable`](https://docs.oracle.com/javase/8/docs/api/java/lang/Iterable.html) which lazily loads and
+returns the data as a user iterates through the results. In the case of Accumulo, this means a connection to Accumulo must
+remain open whilst this iteration takes place. This should be closed automatically when the end of the results is reached.
+However, if you decide not to read all the results, i.e. you just want to check if the results are not empty -
+`!results.iterator().hasNext()` - or an exception is thrown whilst iterating, then the results iterable will not be closed
+and the connection to Accumulo will remain open. Therefore, to be safe results should always be consumed in a
+[try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) block.
+
+**Following on from the previous question, why can't I iterate through results in parallel?**
+
+As mentioned above, the results iterable holds a connection open to Accumulo. To avoid opening multiple connections and
+accidentally leaving the connections open, the Accumulo Store only allows one iterator to be active at any time. When you
+call `.iterator()` the connection is opened. If you call `.iterator()` again, the original connection is closed and a new
```

Comment:
I'm not sure either. I think this just means that only one instance of an `iterator` can be open at a time. As you suggest, to do parallel processing you could use a `spliterator` which would not make multiple `Iterator` objects. It feels like this might be to warn against creating separate `Iterator` objects per thread, or something like that.

Reply:
That's fair probably valid to warn against calling `iterable()` more than once. However, where it mentions using the streams API not being possible I think it is incorrect as you can construct a stream using a `spliterator` on the existing `Iterable` to do both parallel or sequential computation.

Reply:
I've reworded to mention that sometimes it might not be possible to use the Streams API. That should be ok, unless this will always be possible?

Reply:
Yeah think that makes sense, it kind of depends on the construction of the stream making the user aware of it is the main thing